### DISCUSSION TITLE
Background Music v0.4.0

### DISCUSSION THREAD

**Discussion by kyleneideck** [kyleneideck/BackgroundMusic#605] [General]
If you don't know which file to download, you want `BackgroundMusic-0.4.0.pkg` and you can ignore the others.

Changes since v0.3.2:

- Make sure the installer launches BGMApp as the logged-in user. (4c7eba30afb7c19567e1920dc20d9f5558b7da37)
- Add app volume workaround for Google Chrome. See #61. (dfad77dc35cd065390cff0d870ed6bcb81042d7e)
- BGMDriver: Copy settings when a new client matches an existing one. (f61f998c5142f05fc26f412327d28bd4d7e8aece)
- Fix stale data when recording system audio #217 (40f0128dcd5319da6f1a1795c1b733a26c6fe675)
- Add optional debug logging in release builds. (2c1677305d5ce9b932965be4e4cad4187ab83610)
- Hopefully fix a segfault caused by BGM_PlayThrough. (488ed16dfce7950ac6d0aef2cd904abbe12e097e)
- Fix apps failing to start audio after upgrading Background Music. (70a64de73428820e345dbfb5b33ec51b5267afa5)
- Try to avoid using the discrete GPU on multi-GPU systems to save power. (f24a6cb2f12b35820dfc825d58d7340b72f618ba)
- Workaround deadlocks when starting IO on macOS Big Sur. (c0ab98b98b0f2fab1f1fb4ddf1b19c807e71c972)
- Added app volume and pan setting to AppleScript #457 (9765193c1fb42d28bbfc1d881424daf554e999a6)
- Add a bundle ID property for apps in the AppleScript API. (213c339309b4f65206e82a388375ed73ddc5af13)
- Installer: Retry if opening BGMApp fails. (0e0cc5fd05cf14454137ac09424897a2054546f5)
- Fix for issue #475 dropdown arrows move instead of doing something (24b248e09d727db5fdf1ca1a33efc839d537f1e7)
- Fix installer errors on macOS Monterey due to removal of Python 2 (01217d2d8c4a2a7e4fbe0c7d8e2864aa91a13fb1)
- Add app volume workaround for Microsoft Edge #597 (4608f4bbac3c1e6a1609efd596e3f021fae6bb9d)
- Installer: Fix post-install check for BGMDriver failing unnecessarily. (b2770f1793680150fa6138ed75c3a9bc5406d997)
- Show an error message if the user denies input device permissions. (89eebb0e021a8b438869ab9e563f19565e189299)
- Fix slider knob sizes. (d0482871409f1f90d16e0ec5b8e859f3ac6e03dc)
- Drop support for macOS 10.9. (d0482871409f1f90d16e0ec5b8e859f3ac6e03dc)

<hr /><em>This discussion was created from the release <a href='https://github.com/kyleneideck/BackgroundMusic/releases/tag/v0.4.0'>Background Music v0.4.0</a>.</em>

**Comment by vivekp6300** [comment]
The installer (.pkg) file is failing on Sonoma 14.4.1 running on M2 Max Chip.
Since this is GUI version I do not get any informative message - simply that the installer has failed

**Reply by Wirone** [reply]
EDIT: It was installed correctly from CLI after installing system upgrades and restarting the computer 👍.

-----

~~Similar here, Sonoma 14.4 on M1 Pro. Installing from CLI does not give meaningful information too:~~

```shell
sudo installer -verbose -pkg ~/Desktop/BackgroundMusic-0.4.0.pkg -target /
installer: Package name is Background Music 0.4.0
installer: Upgrading at base path /
installer: Preparing for installation….....
installer: Preparing the disk….....
installer: Preparing Background Music 0.4.0….....
installer: Waiting for other installations to complete….....
installer: Configuring the installation….....
installer: Moving items into place….....
installer: Running package scripts…..... # ← Multiplied many times, I just removed superfluous lines
installer: Validating packages….....
installer: The upgrade failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “BackgroundMusic-0.4.0.pkg”.)
```
